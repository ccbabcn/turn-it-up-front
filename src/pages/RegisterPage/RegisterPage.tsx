import Footer from "../../components/Footer/Footer";
import FormRegister from "../../components/forms/FormRegister/FormRegister";
import Header from "../../components/Header/Header";
import { PageStyles } from "../PageStyles";

const RegisterPage = (): JSX.Element => {
  return (
    <>
      <Header />
      <PageStyles>
        <h2>REGISTER</h2>
        <FormRegister />
      </PageStyles>
      <Footer />
    </>
  );
};

export default RegisterPage;
