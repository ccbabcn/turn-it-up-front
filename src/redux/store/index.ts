import { configureStore } from "@reduxjs/toolkit";
import uiReducer from "../features/uiSlice/uiSlice";
import userReducer from "../features/userSlice/userSlice";

const store = configureStore({
  reducer: {
    user: userReducer,
    ui: uiReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
