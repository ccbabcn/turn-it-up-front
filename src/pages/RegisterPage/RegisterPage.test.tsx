import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import store from "../../redux/store";
import RegisterPage from "./RegisterPage";
import reactTestRenderer from "react-test-renderer";

describe("Given a RegisterPage component", () => {
  describe("When invoked", () => {
    test("Then it should render a heading with the text 'REGISTER'", () => {
      render(
        <Provider store={store}>
          <BrowserRouter>
            <RegisterPage />
          </BrowserRouter>
        </Provider>
      );

      const renderedHeading = screen.getByRole("heading", {
        name: /REGISTER/i,
      });

      expect(renderedHeading).toBeInTheDocument();
    });
  });
  describe("When the component is rendered", () => {
    test("Then it should match it's snapshot", () => {
      const loginPageSnap = reactTestRenderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <RegisterPage />
          </BrowserRouter>
        </Provider>
      );

      expect(loginPageSnap).toMatchSnapshot();
    });
  });
});
