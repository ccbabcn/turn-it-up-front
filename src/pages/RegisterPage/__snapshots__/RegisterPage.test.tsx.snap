// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a RegisterPage component When the component is rendered Then it should match it's snapshot 1`] = `
<div
  className="sc-gsnTZi kscNpA"
>
  <h2>
    REGISTER
  </h2>
  <div
    className="sc-bczRLJ hmTf"
  >
    <form
      autoComplete="off"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="formField"
      >
        <label
          htmlFor="name"
        >
          Name
        </label>
        <input
          id="name"
          onChange={[Function]}
          placeholder="Name"
          type="text"
          value=""
        />
      </div>
      <div
        className="formField"
      >
        <label
          htmlFor="username"
        >
          Username
        </label>
        <input
          id="username"
          onChange={[Function]}
          placeholder="Username"
          type="text"
          value=""
        />
      </div>
      <div
        className="formField"
      >
        <label
          htmlFor="password"
        >
          Password
        </label>
        <input
          id="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <button
        className="form-button"
        disabled={false}
        type="submit"
      >
        REGISTER
      </button>
      <div
        className="form-options"
      >
        Already have an account? 
        <a
          className=""
          href="/login"
          onClick={[Function]}
        >
          Login now
        </a>
      </div>
    </form>
  </div>
</div>
`;
